src/pages/deposit.tsx

```diff
@@ -1,154 +1,189 @@
+  function tog_center() {
+    setmodal_deposit(!modal_deposit)
+    removeBodyCss()
+  }
 
-    const handleDeposit = async () => {
-      // console.log( BNtoNum(Number(0.21), 18));
-      // console.log(NumToBN(Number(0.21), 18));
-      try {
-        setIsTransactionDone(true);
-        const approveTransactionHash = await wrapper?.getMockBep20Instance().approve(SymbolsMap[props.asset], inputVal, DecimalsMap[props.asset]);
-        console.log("Approve Transaction sent: ", approveTransactionHash);
-        const _commitPeriod: string | undefined =  commitPeriod;
-        
-        const tx = await wrapper?.getDepositInstance().depositRequest(SymbolsMap[props.asset], CommitMap[_commitPeriod], inputVal, DecimalsMap[props.asset]);
-        console.log(tx);
-        onDeposit(tx.events);
-      } catch (err) {
-        setIsTransactionDone(false);
-        if (err instanceof Object) {
-          toast.error(`${GetErrorText(String(err['message']))}`, { position: toast.POSITION.BOTTOM_RIGHT, closeOnClick: true});
-        } else {
-          toast.error(`${GetErrorText(String(err))}`, { position: toast.POSITION.BOTTOM_RIGHT, closeOnClick: true});
-        }
-      }
-    }
+  const handleDeposit = async () => {
+    // console.log( BNtoNum(Number(0.21), 18));
+    // console.log(NumToBN(Number(0.21), 18));
+    try {
+      setIsTransactionDone(true)
+      const approveTransactionHash = await wrapper
+        ?.getMockBep20Instance()
+        .approve(SymbolsMap[props.asset], inputVal, DecimalsMap[props.asset])
+      await approveTransactionHash.wait()
+      console.log("Approve Transaction sent: ", approveTransactionHash)
+      const _commitPeriod: string | undefined = commitPeriod
 
-    const onDeposit = (data) => {
-      setIsTransactionDone(false);
-      const res = data['DepositAdded'] ? data['DepositAdded']['returnValues'] : data['NewDeposit']['returnValues'];
-      let amount = BNtoNum(Number(res.amount),DecimalsMap[res.market]);
-      toast.success(`Deposited amount: ${amount}`, { position: toast.POSITION.BOTTOM_RIGHT, closeOnClick: true});
+      const tx1 = await wrapper
+        ?.getDepositInstance()
+        .depositRequest(
+          SymbolsMap[props.asset],
+          CommitMap[_commitPeriod],
+          inputVal,
+          DecimalsMap[props.asset]
+        )
+      const tx = await tx1.wait()
+      onDeposit(tx.events)
+    } catch (err) {
+      setIsTransactionDone(false)
+      toast.error(`${GetErrorText(err)}`, {position: toast.POSITION.BOTTOM_RIGHT, closeOnClick: true,})
     }
+  }
 
+  const onDeposit = data => {
+    let eventName
+    let _amount
+    data.forEach(e => {
+      if (e.event == "DepositAdded" || e.event == "NewDeposit") {
+        eventName = e.event
+        _amount = e.args.amount.toBigInt()
+      }
+    })
+    setIsTransactionDone(false)
+    // const res = data['DepositAdded'] ? data['DepositAdded']['returnValues'] : data['NewDeposit']['returnValues'];
+    let amount = BNtoNum(_amount)
```

Comment:
decimal parameter is missing. Also in toaster its showing wrong representation of amoun.

Reply:
FIxed in BEP20DecimalHandling branch